internal/olm/operator/registry/fbcindex/fbc_registry_pod.go

```diff
@@ -208,6 +209,13 @@ func (f *FBCRegistryPod) podForBundleRegistry(cs *v1alpha1.CatalogSource) (*core
 			Namespace: f.cfg.Namespace,
 		},
 		Spec: corev1.PodSpec{
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: pointer.Bool(true),
+				RunAsUser:    pointer.Int64(1001),
```

Comment:
Do we need the RunAsUser?
If we use it then, this container cannot run as resctricted for OCP
Could we remove the RunAsUser from here?
Is not the Dockerfile of this image defining a user already?
@rashmigottipati

Reply:
@camilamacedo86 I thought that as long as the `RunAsUser` value is non-zero numeric that it would work when `RunAsNonRoot` is set to `true`. What is the difference that makes it so it wouldn't run as restricted in OCP?

Reply:
@everettraven that is valid ONLY for k8s.
Also, add RunAsUser is not a requirement at all for the container be qualified as restricted.

⚠️ With some configurations, your solution might comply with the restricted Kubernetes definition but will not be accepted under the SCC restricted-v2 in OCP. In K8s you can specify the runAsUser and get the Pod/Container running in a restricted namespace however, for Openshift’s restricted/restricted-v2 SCC you MUST leave the runAsUser field empty, or provide a value that falls within the specific user range for the namespace. Otherwise, you will only be able to run the Pod if it has access to the SCC nonroot-v2 or anyuid.

So, could we please remove the runAsUser from here?

/hold

Reply:
Hmm. I think in the case of the default image it should be fine, but I think that our concern is that if a user provides an index image that the user is root(0) then we will run into an issue. I think for now removing the `RunAsUser` field should be fine, but I wonder if there is a way for us to get the user range for a namespace?

Reply:
I removed this field for now. I will document the limitation that if the user provides an index image with the default user in the container as root(0) then it will most likely fail without this field.